composer.json

```diff
@@ -8,19 +8,27 @@
             "name": "Georg Ehrke"
         }
     ],
-    "require": {},
     "require-dev": {
         "phpunit/phpunit": "^9.5",
-        "nextcloud/coding-standard": "^0.5.0"
+		"christophwurst/nextcloud": "dev-master@dev",
+		"psalm/phar": "^4.22",
+		"nextcloud/coding-standard": "^1.0"
     },
     "scripts": {
         "lint": "find . -name \\*.php -not -path './vendor/*' -print0 | xargs -0 -n1 php -l",
         "cs:check": "php-cs-fixer fix --dry-run --diff",
         "cs:fix": "php-cs-fixer fix",
+        "psalm": "psalm.phar --threads=1",
+		"psalm:update-baseline": "psalm.phar --threads=1 --update-baseline --set-baseline=tests/psalm-baseline.xml",
+        "psalm:clear": "psalm.phar --clear-cache && psalm --clear-global-cache",
+        "psalm:fix": "psalm.phar --alter --issues=InvalidReturnType,InvalidNullableReturnType,MissingParamType,InvalidFalsableReturnType",
         "test": "phpunit --configuration phpunit.xml --fail-on-warning",
         "test:integration": "phpunit --configuration phpunit.integration.xml --fail-on-warning"
     },
     "config": {
+        "allow-plugins": {
+            "composer/package-versions-deprecated": true
+        },
         "platform": {
             "php": "7.4"
```

Comment:
same here - 8.0 is recommended for current nextcloud release, should be fine /w psalm >=4.x